### DISCUSSION TITLE
Why is `Table.join()` not creating hierarchies as expected?

### DISCUSSION THREAD

**Discussion by TravisWhip** [atoti/atoti#504] [Q&A] [answered]
Hello, I have tried to create a cube with two tables joined together (Table Actors and Table Movies). I join the two tables together (`Actors.join(Movies)`), then load data into the two tables, then finally create a cube with the base table as Actors. When I look at the cube I do no see any hierarchies from the joined Movies table.

I have also tried to create just the Actors table, load data and create the Cube with Actors as base table. Then create and load a Movies table then join it to the Actors table.

I would expect in either case to have the levels from the Movies table to show up in the Cube.

Attached is a Zip of my notebook containing all input data:
[Atoti_Actors_Movies.zip](https://github.com/atoti/atoti/files/8198560/Atoti_Actors_Movies.zip)

**Comment by tibdex** [comment, marked as answer]
From the docs of [`Table.join()`](https://docs.atoti.io/0.6.4/lib/atoti.table.html#atoti.table.Table.join):

<img width="1277" alt="Screen Shot 2022-03-07 at 16 32 31" src="https://user-images.githubusercontent.com/6574550/157065200-dae95b3c-1aca-4b13-b365-aec8fb4ae984.png">

And your schema:

<img width="468" alt="Screen Shot 2022-03-07 at 16 33 53" src="https://user-images.githubusercontent.com/6574550/157065421-ea74f00d-badf-4bf2-ad0f-0475ec7b8346.png">

It's expected that **Movies** does not add hierarchies to the cube since:
- **Movie** is a key column
- **awards_won** is a numeric column.

To force the creation of an **awards_won** hierarchy you could do:

```python
h["Awards won"] = [movie_awards_table["awards_won"]]
```

**Reply by TravisWhip** [reply]
Okay so if I use the `mode="manual"` when creating the cube I should expect to have an **awards_won** hierarchy after joining the two stores together?

**Reply by tibdex** [reply]
> `manual`: The un-mapped keys of the other table will become hierarchies.

**awards_won** is not a key so, by extension, it's not an "un-mapped key" and thus won't automatically become a hierarchy.
